python/ml4ir/model/ranking_model.py

```diff
@@ -152,11 +160,21 @@ def build_model(
         # Create model with functional Keras API
         model = Model(inputs=inputs, outputs={"ranking_scores": predictions})
 
+        # Get loss and metrics
+        loss_fn = loss.get_loss_fn(mask=metadata_features["mask"])
+        metric_fns: List[kmetrics.Metric] = list()
```

Comment:
just do this as a for-comprension, no?

`metric_fns: List[kmetrics.Metric] = [metric_factor.get_metric_impl(metric, ...) for metric in metrics]`

Reply:
I can. It will be long and dirty. I favored clean code here as opposed to memory(very minimal). Also, I am doing an `extend` and not an `append`. So it will be nested list comprehension. Again, doable - just ugly. Let me know if you think that is better.

Reply:
oh get_metric_impl() returns a list... hence you'd really like a flatMap method :\

ok, so this *could* be done with a `reduce()` call.  But no, it wouldn't be any more readable than what you've got.